File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3

"""Write an empty class Rectangle that defines a rectangle:"""


class Rectangle:
    """define the self attributes"""
    def __init__(self, width=0, height=0):
        self.width = width
        self.height = height

    @property
    def height(self):
        """get the height"""
        return self.__height

    @height.setter
    def height(self, value):
        """set the height"""
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        elif value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

    @property
    def width(self):
        """get the width"""
        return self.__width

    @width.setter
    def width(self, value):
        """set the width"""
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        elif value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    def area(self):
        return (self.__width * self.__height)

    def perimeter(self):
        if self.__width > 0 and self.__height > 0:
            return ((self.__width * 2) + (self.__height * 2))
        else:
            return 0

    def __str__(self):
        """print the rectangle with the character #"""

        string = ""
        for row in range(self.__height):
            for column in range(self.__width):
                string += '#'

            if self.__width is not 0 and row < (self.__height - 1):
                string += "\n"
        return string

    def __repr__(self):
        """to recreate a new instance by using"""
        return "Rectangle({:d}, {:d})".format(self.width, self.height)

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node

/*
 * Write an empty class Rectangle that defines a rectangle
 */

class Rectangle {
  constructor (w, h) {
    if ((h > 0 && !isNaN(h)) && (w > 0 && !isNaN(w))) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    for (let i = 0; i < this.height; i++) {
      let row = '';
      for (let j = 0; j < this.width; j++) {
        row += 'X';
      }
      console.log(row);
    }
  }

  rotate () {
    let a = this.width
    this.width = this.height;
    this.height = a;
  }

  double () {
    this.width = this.width * 2;
    this.height = this.height * 2;
    return this.width * this.height * 2;
  }
}
module.exports = Rectangle;
